x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/DateRangeBlock.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.ReleasableIterator;
+
+/**
+ * Block that stores aggregate_metric_double values.
+ */
+public sealed interface DateRangeBlock extends Block permits DateRangeArrayBlock, ConstantNullBlock {
```

Comment:
I don't see anything in this block that is Date or DateRange specific. It is basically a block of two long blocks. Should it not be called something more generic, like LongLongBlock, or TwoLongsBlock?